src/main/scala/com/mongodb/spark/MongoConnector.scala

```diff
@@ -228,6 +230,12 @@ case class MongoConnector(mongoClientFactory: MongoClientFactory)
     withCollectionDo[D, T](config, { collection => code.call(collection) })
   }
 
+  private[spark] def hasSampleAggregateOperator(readConfig: ReadConfig): Boolean = {
```

Comment:
It's an edge case, but it's possible that we could mis-detect $sample support is a mixed-version cluster.   I'd suggest a note in the docs.